packages/components/nodes/retrievers/MeilisearchRetriever/MeilisearchRetriever.ts

```diff
@@ -0,0 +1,177 @@
+import { Embeddings } from '@langchain/core/embeddings'
+
+class MeilisearchRetriever_node implements INode {
+    label: string
+    name: string
+    version: number
+    description: string
+    type: string
+    icon: string
+    category: string
+    baseClasses: string[]
+    inputs: INodeParams[]
+    credential: INodeParams
+    badge: string
+    outputs: INodeOutputsValue[]
+
+    constructor() {
+        this.label = 'Meilisearch retriever'
+        this.name = 'Meilisearch retriever'
+        this.version = 1.0
+        this.type = 'Meilisearch'
+        this.icon = 'Meilisearch.png'
+        this.category = 'Vector Stores'
+        this.badge = 'NEW'
+        this.description = 'Meilisearch uses hybrid search to sematically answer the query.'
+        this.baseClasses = ['BaseRetriever']
+        this.credential = {
+            label: 'Connect Credential',
+            name: 'credential',
+            type: 'credential',
+            credentialNames: [this.type, 'MeilisearchApi']
+        }
+        this.inputs = [
+            {
+                label: 'Document',
+                name: 'document',
+                type: 'Document',
+                list: true,
+                optional: true
+            },
+            {
+                label: 'Embeddings',
+                name: 'embeddings',
+                type: 'Embeddings'
+            },
+            {
+                label: 'host',
+                name: 'host',
+                type: 'string',
+                description: 'This is the URL for the desired Meilisearch instance'
+            },
+            {
+                label: 'indexUid',
+                name: 'indexUid',
+                type: 'string',
+                description: 'UID for the index to answer from'
+            },
+            {
+                label: 'Top K',
+                name: 'K',
+                type: 'number',
+                description: 'number of top searches to return as context',
+                additionalParams: true,
+                optional: true
+            },
+            {
+                label: 'Semantic Ratio',
+                name: 'semanticRatio',
+                type: 'number',
+                description: 'percentage of sematic reasoning in meilisearch hybrid search',
+                additionalParams: true,
+                optional: true
+            }
+        ]
+        this.outputs = [
+            {
+                label: 'Meilisearch Retriever',
+                name: 'meilisearch retriever',
+                description: 'retrieve answers',
+                baseClasses: this.baseClasses
+            },
+            {
+                label: 'another output',
+                name: 'meilisearch documents',
+                description: 'return documents containing pageContent and metadata',
+                baseClasses: ['Document', 'json']
+            }
+        ]
+    }
+    //@ts-ignore
+    vectorStoreMethods = {
```

Comment:
is this a retriever or a vector store? Because upsert only for vector store

Reply:
well , this component is similar to postgres component , it is essentially a vectorstore (meilisearch index that contain embedded documents) , but also , it is a retriever that utilize hybrid search to retrieve context from the meilisearch index (which was embedded using an external embeddings) , so yes the component does have upsert functionality